src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -3033,7 +3033,7 @@ private static bool IsSignedIntegralType(TypeSymbol type)
                 case SpecialType.System_Int16:
                 case SpecialType.System_Int32:
                 case SpecialType.System_Int64:
-                case SpecialType.System_IntPtr:
+                case SpecialType.System_IntPtr when type.IsNativeIntegerTypeOrNumericIntPtr:
```

Comment:
>when type.IsNativeIntegerTypeOrNumericIntPtr [](http://example.com/codeflow?start=47&length=44)

Was there a bug here previously for `nint`?

Reply:
In the technical sense, yes, but it's not observable with our standard definitions of IntPtr/UIntPtr. 
Overload resolution prefers signed parameter to unsigned ones, but that logic should only apply to numeric types.
But currently there's not type that would implicitly convert to IntPtr/UIntPtr with our standard definitions (the conversions from `int` or `void*` are explicit).
Still, I thought it's better for correctness.